test/lint-variable-use-def.js

```diff
@@ -0,0 +1,371 @@
+      await assertLintFree(
+        `
+          <emu-clause id="sec-valid-chosen-reads">
+            <h1>Valid Chosen Reads</h1>
+            <p>A candidate execution _execution_ has valid chosen reads if the following algorithm returns *true*.</p>
+            <table>
+              <tr>
+              <td>
+              <emu-alg>
+                1. If some condition of _execution_, return *true*.
+                1. Return *false*.
+              </emu-alg>
+              </td>
+              <td>
+              <emu-alg>
+                1. If some condition of _execution_, return *true*.
+                1. Return *false*.
+              </emu-alg>
+              </td>
+              </tr>
+            </table>
+          </emu-clause>
+        `
+      );
+    });
+
+    it('the receiver for concrete methods is visible', async () => {
+      await assertLintFree(
+        `
+          <emu-clause id="sec-declarative-environment-records-hasbinding-n" type="concrete method">
+            <h1>HasBinding ( _N_ )</h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Declarative Environment Record _envRec_</dd>
+            </dl>
+            <emu-alg>
+              1. If _envRec_ has a binding for the name that is the value of _N_, return *true*.
+              1. Return *false*.
+            </emu-alg>
+          </emu-clause>
+        `
+      );
+    });
+
+    it('abstract closure parameters/captures are visible', async () => {
+      await assertLintFree(
+        `
+          <emu-clause id="sec-object.fromentries">
+          <h1>Object.fromEntries ( _obj_ )</h1>
+          <emu-alg>
+            1. Let _closure_ be a new Abstract Closure with parameters (_key_, _value_) that captures _obj_ and performs the following steps when called:
+              1. Do something with _obj_, _key_, and _value_.
+              1. Return *undefined*.
+            1. Return _closure_.
+          </emu-alg>
+        </emu-clause>
+        `
+      );
+    });
+
+    it('abstract closure parameters/captures are visible', async () => {
+      await assertLintFree(
+        `
+          <emu-clause id="sec-object.fromentries">
+          <h1>Object.fromEntries ( <del>_del_</del><ins>_obj_</ins> )</h1>
+          <emu-alg>
+            1. Let _closure_ be a new Abstract Closure with parameters (_key_, _value_) that captures _obj_ and performs the following steps when called:
+              1. Do something with _obj_, _key_, and _value_.
+              1. Return *undefined*.
+            1. Return _closure_.
+          </emu-alg>
+        </emu-clause>
+        `
+      );
+    });
+
+    it('multiple declarations are visible', async () => {
+      await assertLintFree(
+        `
+          <emu-alg>
+            1. Let _x_, _y_, and _z_ be some values.
+            1. Return _x_ + _y_ + _z_.
+          </emu-alg>
+        `
+      );
+    });
+
+    it('"such that" counts as a declaration"', async () => {
```

Comment:
```suggestion
    it('"such that" counts as a use"', async () => {
```

Reply:
Nope, it's a declaration. For example

```
While _finalizationRegistry_.[[Cells]] contains a Record _cell_ such that _cell_.[[WeakRefTarget]] is ~empty~, [...]
```
declares `_cell_`.